doc/pfmatch.md

```diff
@@ -0,0 +1,234 @@
+# Pfmatch
+
+Pfmatch is a pattern-matching language for network packets, embedded in
+Lua.  It is built on the well-known
+[pflang](https://github.com/Igalia/pflua/blob/master/doc/pflang.md)
+packet filtering language, using the fast
+[pflua](https://github.com/Igalia/pflua/blob/master/README.md) compiler
+for LuaJIT
+
+Here's an example of a simple pfmatch program that just divides up
+packets depending on whether they are UDP, TCP, or something else:
+
+```lua
+match {
+   tcp => handle_tcp
+   udp => handle_udp
+   otherwise => handle_other
+}
+```
+
+Unlike pflang filters written for such tools as `tcpdump`, a pfmatch
+program can dispatch packets to multiple handlers, potentially
+destructuring them along the way.  In contrast, a pflang filter can only
+say "yes" or "no" on a packet.
+
+Here's a more complicated example that passes all non-IP traffic, drops
+all IP traffic that is not going to or coming from certain IP addresses,
+and calls a handler on the rest of the traffic.
+
+```lua
+match {
+   not ip => forward
+   -- Drop fragmented packets.
+   ip[6:2] & 0x3fff != 0 => drop
+   ip src 1.2.3.4 => incoming_ip
+   ip dst 5.6.7.8 => outgoing_ip
+   otherwise => drop
+}
+```
+
+In the example above, the handlers after the arrows (`=>`) are Lua
+functions.  If a handler matches (more on that later), it will be called
+with two arguments: the packet data and the length.  You can pass more
+arguments by specifying them after the handler.  For example, we could
+pass the offset of the start of the IP header by using the [address-of
+extension](https://github.com/Igalia/pflua/blob/master/doc/extensions.md):
+
+
+```lua
+match {
+   not ip => forward
+   -- Drop fragmented packets.
+   ip[6:2] & 0x3fff != 0 => drop
+   ip src 1.2.3.4 => incoming_ip(&ip[0])
+   ip dst 5.6.7.8 => outgoing_ip(&ip[0])
+   otherwise => drop
+}
+```
```

Comment:
Dunno, you don't find the two sentences preceding the example as sufficiently explanatory?